lock-api-objects/src/main/java/com/palantir/lock/watch/LockEvent.java

```diff
@@ -52,6 +66,63 @@ public <T> T accept(Visitor<T> visitor) {
         return visitor.visit(this);
     }
 
+    @JsonCreator
+    public static LockEvent create(
+            @JsonProperty("sequence") long sequence,
+            @JsonProperty("lockDescriptors") List<LockDescriptor> lockDescriptors,
+            @JsonProperty("lockToken") LockToken lockToken,
+            @JsonProperty("metadata") Optional<WireLockRequestMetadata> optWireMetadata) {
+        Optional<LockRequestMetadata> metadata = optWireMetadata.map(wireMetadata -> {
+            KeyListChecksum checksum = KeyListChecksum.of(
+                    ChecksumType.valueOf(wireMetadata.lockListChecksum().typeId()),
+                    wireMetadata.lockListChecksum().value());
+            IndexEncodingResult<LockDescriptor, ChangeMetadata> indexEncodingResult =
+                    IndexEncodingResult.<LockDescriptor, ChangeMetadata>builder()
+                            .keyList(lockDescriptors)
+                            .indexToValue(wireMetadata.indexToChangeMetadata())
+                            .keyListChecksum(checksum)
+                            .build();
+            Map<LockDescriptor, ChangeMetadata> lockDescriptorToChangeMetadata =
+                    IndexEncodingUtils.decode(indexEncodingResult, Function.identity());
```

Comment:
This is long enough that I think it would read easier if this was factored into a separate method (and then you can just call the parameter wireMetadata without the opt prefix!)

Reply:
Since we need the `lockDescriptors` argument in the conversion, we'd have to do some currying in order to achieve this (which I am not against).